### PR TITLE
[Pre-release] Enable bulk invitations on onboarding step

### PR DESCRIPTION
**Pull request by gazayas**
Joint PR:
- https://github.com/bullet-train-co/bullet_train/pull/835

See the joint PR https://github.com/bullet-train-co/bullet_train/pull/835 for additional information.

Original PR description:

Currently looks like this, I definitely have a lot more work to do on it though.

スクリーンショット 2023-08-10 21 28 53

### CODE REVIEW

**Comment by gazayas** [comment]
I'm currently stuck on this one, specifically with how to apply strong parameters. We have to accept nested attribute for each Invitation AND a Membership. I'm sure it's something small I'm missing, just having some trouble with this one.

```
Unpermitted parameters: :authenticity_token, :account_onboarding_invitation_list, :commit. Context: { controller: Account::Onboarding::InvitationListsController, action: create, request: #<ActionDispatch::Request:0x000000011b1738f8>, params: {"authenticity_token"=>"[FILTERED]", "account_onboarding_invitation_list"=>{"team_id"=>"1", "creator_membership_id"=>"1", "invitations_attributes"=>{"0"=>{"email"=>"test@test.com", "membership_attributes"=>{"role_ids"=>"Foo"}}, "1"=>{"email"=>"tester@tester.com", "membership_attributes"=>{"role_ids"=>"Foo"}}, "2"=>{"email"=>"testing@testing.com", "membership_attributes"=>{"role_ids"=>"Foo"}}}}, "commit"=>"Next", "controller"=>"account/onboarding/invitation_lists", "action"=>"create"} }
Completed 400 Bad Request in 10ms (ActiveRecord: 2.0ms | Allocations: 4298)


  
ActionController::ParameterMissing (param is missing or the value is empty: {:invitations_attributes=>[:email, {:membership_attributes=>[{:role_ids=>[]}]}]}):
  
actionpack (7.0.7) lib/action_controller/metal/strong_parameters.rb:477:in `require'
local/bullet_train-core/bullet_train/app/controllers/concerns/account/onboarding/invitation_lists/controller_base.rb:29:in `invitation_list_params'
cancancan (3.5.0) lib/cancan/controller_resource_sanitizer.rb:10:in `sanitize_parameters'
```

**Comment by gazayas** [comment]
Currently now just getting this error
スクリーンショット 2023-08-15 17 15 06

**Comment by gazayas** [comment]
Skip button added

**Comment by gazayas** [comment]
Taking a step back from it, we probably don't even need the new model. Will try to do something similar to `Invitation#resend`, although I'll have to play by ear how validations will work out.

**Comment by olbrich** [comment]
I expect that users will pretty quickly want to be able to upload a CSV/XLSX of names and emails to invite, particularly if you are dealing with larger teams.

**Comment by gazayas** [comment]
@olbrich Seems like something we could potentially do with Action Models, will try to keep that in the back of my mind.

**Comment by gazayas** [comment]
Finding that it's hard to do this one without the existence of a model since Bullet Train fields rely on the existence of a form:
```ruby
<%= render 'shared/fields/email_field', form: invitation_form, method: :email, options: {autofocus: true} %>
```

Still running into the strong_params issue, so I think I'll take a step back from this one for a bit to clear my head.

**Comment by gazayas** [comment]
Made a little more progress on this one. Now we can see if the Invitations and Memberships are valid or not, and it will redirect with an error saying "Please correct the errors below" if something's wrong.

Will have to actually send the invitations next and work on the system tests.

**Comment by gazayas** [comment]
I have a bunch of hidden fields for now, but I plan on just moving this logic to the controller.

**Comment by gazayas** [comment]
@andrewculver @jagthedrummer The tests are now passing!

This one is ready for review.

File: bullet_train/app/controllers/concerns/account/onboarding/invitation_lists/controller_base.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    # Set default values for invitations and memberships.
+    # `save` below checks if the values are valid or not.
+    @account_onboarding_invitation_list.team = current_team
+    @account_onboarding_invitation_list.invitations.each_with_index do |invitation, idx|
+      invitation.team = current_team
+      invitation.from_membership = current_membership
+      invitation.membership.team = current_team
+
+      # Role IDs don't get registered automatically because roles_ids is an array, so we handle that here.
+      if available_roles.include?(params[:account_onboarding_invitation_list][:invitations_attributes][idx.to_s][:membership_attributes][:role_ids].downcase)
```
Comment (gazayas) [inline bullet_train/app/controllers/concerns/account/onboarding/invitation_lists/controller_base.rb:28]: We could probably make the params call here shorter, will try to keep this one in the back of my mind.

File: bullet_train/app/controllers/concerns/account/onboarding/invitation_lists/controller_base.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
+module Account::Onboarding::InvitationLists::ControllerBase
+  extend ActiveSupport::Concern
+
+  included do
+    layout "devise"
+
+    before_action do
+      @user = current_user
```
Comment (gazayas) [inline bullet_train/app/controllers/concerns/account/onboarding/invitation_lists/controller_base.rb:8]: Thinking we might need to do something else for this

**Pull request merged**